### PR TITLE
Tweakable PID from Simulation

### PR DESCRIPTION
**Pull request by smit585**
Hey Thomas,

Just wanted to get your opinion on this. It doesn't necessarily need to merge.

Pressing 'P' will increase Kp by 1
Pressing 'I' will increase Ki by 0.1
Pressing 'D' will increase Kd by 0.1

Small visual notification shade I added just so the text is more visual. And few progress bar to show throttle and steer values. 

Also, it needs to be checked whether this code runs or not. I can not run the code the way you have mentioned in the script.  
I had made functions that could tweek the constants and a help box to display all the options, but could not bring the changes here as I am not able to run via this code. 

For this, I had created a different project in Pycharm in which I made the edits and then brought them here. 
Do let me know how you have set up your environment. 

Thanks
Smit Sparsh

### CODE REVIEW

**Comment by thomasfermi** [comment]
Hey Smit, thank you so much for your pull request :)! I just read the code and it looks very nice. Only issue I spotted at first glance is that you removed a function from carla_util.py that is being used in the chapter on lane detection. 

I will hopefully find time this weekend to actually run and test your code on my system.

File: code/tests/control/carla_sim.py [hunk 1]
```diff
@@ -161,7 +162,7 @@ def main(use_lane_detector=False, ex=False):
         # Create a synchronous mode context.
         with CarlaSyncMode(world, *sensors, fps=FPS) as sync_mode:
             while True:
-                if should_quit():
```
Comment (smit585) [inline code/tests/control/carla_sim.py:164]: Yeah, I think I should not have changed the name of this function. Sorry about that. But for some reason creating another function that again checks the pygame event did not work as expected.

**Comment by smit585** [comment]
I have closed the pull request. I am finally able to run the code via anaconda. Will create a new one soon!

**Comment by thomasfermi** [comment]
Understood. Looking forward to the new Pull Request! :)

**Pull request closed without merging**